src/inmanta/server/agentmanager.py

```diff
@@ -290,21 +290,38 @@ async def resume_agents(self, env: data.Environment, connection: Optional[asyncp
         await self._unpause_agent(env, endpoint=const.AGENT_SCHEDULER_ID, connection=connection)
 
     @handle(methods_v2.all_agents_action, env="tid")
-    async def all_agents_action(self, env: data.Environment, action: AgentAction) -> None:
-        if env.halted and action in {AgentAction.pause, AgentAction.unpause}:
+    async def all_agents_action(self, env: data.Environment, action: AllAgentAction) -> None:
+        if env.halted and action in {AllAgentAction.pause, AllAgentAction.unpause}:
             raise Forbidden("Can not pause or unpause agents when the environment has been halted.")
-        if not env.halted and action in {AgentAction.keep_paused_on_resume, AgentAction.unpause_on_resume}:
+        if not env.halted and action in {AllAgentAction.keep_paused_on_resume, AllAgentAction.unpause_on_resume}:
             raise Forbidden("Cannot set on_resume state of agents when the environment is not halted.")
-        if action is AgentAction.pause:
-            await self._pause_agent(env=env)
-        elif action is AgentAction.unpause:
-            await self._unpause_agent(env=env)
-        elif action is AgentAction.keep_paused_on_resume:
-            await self._set_unpause_on_resume(env, should_be_unpaused_on_resume=False)
-        elif action is AgentAction.unpause_on_resume:
-            await self._set_unpause_on_resume(env, should_be_unpaused_on_resume=True)
+        match action:
+            case AllAgentAction.pause:
+                await self._pause_agent(env=env)
+            case AllAgentAction.unpause:
+                await self._unpause_agent(env=env)
+            case AllAgentAction.keep_paused_on_resume:
+                await self._set_unpause_on_resume(env, should_be_unpaused_on_resume=False)
+            case AllAgentAction.unpause_on_resume:
+                await self._set_unpause_on_resume(env, should_be_unpaused_on_resume=True)
+            case AllAgentAction.remove_all_agent_venvs:
+                await self._remove_executor_venvs(env)
+            case _:
+                assert_never(f"Unknown agent action: {action.name}")
+
+    async def _remove_executor_venvs(self, env: data.Environment) -> None:
+        """
+        Remove the venvs of the executors used by the given environment.
+        """
+        agent_client = self.get_agent_client(tid=env.id, endpoint=AGENT_SCHEDULER_ID, live_agent_only=False)
+        if agent_client:
+            result = await agent_client.remove_executor_venvs()
+            if result.code != 200:
+                raise Exception(
+                    f"Failed to remove executor venvs: {result.result['message'] if result.result is not None else ''}"
+                )
```

Comment:
Should we start a background task here instead of doing a blocking wait?

Reply:
I discussed this with Wouter. The plan now is to:

* Make API call asynchronous, because the execution time of a handler can be arbitrary long.
* For now, create a notification when the rebuild starts and when it ends.

Create a ticket for a more elegant solution by having support to request the scheduler status from the server. The web-console can then rely on this to report that deploy operations are currently suspended.

Reply:
I added the proposal to the ticket and added the discussion label, so that we can discuss it during the planning meeting when we get to it.